PennMobile/src/main/java/com/pennapps/labs/pennmobile/GsrFragment.kt

```diff
@@ -0,0 +1,519 @@
+                            gsrHashMap["Education Commons"] = 2495
+                            gsrHashMap["Biomedical Library"] = 2683
+                            gsrHashMap["Fisher Fine Arts"] = 2637
+                            gsrHashMap["Levin Building"] = 1090
+                            gsrHashMap["Museum Library"] = 2634
+                            gsrHashMap["VP Seminar"] = 2636
+                            gsrHashMap["VP Special Use"] = 2611
+
+
+                            gsrLocationsArray.add("Weigle")
+                            gsrLocationsArray.add("VP Ground Floor")
+                            gsrLocationsArray.add("VP 3rd Floor")
+                            gsrLocationsArray.add("VP 4th Floor")
+                            gsrLocationsArray.add("Lippincott")
+                            gsrLocationsArray.add("Education Commons")
+                            gsrLocationsArray.add("Biomedical Library")
+                            gsrLocationsArray.add("Fisher Fine Arts")
+                            gsrLocationsArray.add("Levin Building")
+                            gsrLocationsArray.add("Museum Library")
+                            gsrLocationsArray.add("VP Seminar")
+                            gsrLocationsArray.add("VP Special Use")
+
+
+                            var gsrs = gsrLocationsArray.toTypedArray<String?>()
+
+                            //create an adapter to describe how the items are displayed, adapters are used in several places in android.
+                            //There are multiple variations of this, but this is the basic variant.
+                            activity?.let {activity ->
+                                val adapter = ArrayAdapter(activity, android.R.layout.simple_spinner_dropdown_item, gsrs)
+                                //set the spinners adapter to the previously created one.
+                                gsrDropDown?.setAdapter(adapter)
+                                searchForGSR()
+                            }
+                        }
+                    }
+                }
+                )
+        gsrDropDown?.setOnItemSelectedListener(object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(adapterView: AdapterView<*>, view: View?, i: Int, l: Long) {
+                searchForGSR()
+            }
+
+            override fun onNothingSelected(adapterView: AdapterView<*>) {
+
+            }
+        })
+    }
+
+
+    //helper function that turns military to civilian time
+    fun convertToCivilianTime(input: String): String {
+        val militaryTimeFormatter = DateTimeFormat.forPattern("HH:mm")
+        val civilianTimeFormatter = DateTimeFormat.forPattern("hh:mm a")
+        return civilianTimeFormatter.print(militaryTimeFormatter.withLocale(Locale.ENGLISH).parseLocalTime(input))
+    }
+
+    //helper function that converts string to int but keeps zeros
+    fun safeToString(input: Int): String {
```

Comment:
You can make this a single line function:

`fun safeToString(input: Int): String = if (input == 0) "00" else Integer.toString(input)`

Also is there a reason why we aren't using Joda Time to handle this time parsing?